### PR TITLE
Add command to list pending etchings

### PR DESCRIPTION
**Pull request by ldiego08**
Add a new `ord wallet pending` command to list pending etchings, as mentioned by @raphjaph on #3679.

### CODE REVIEW

File: src/subcommand/wallet/pending.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+#[derive(Debug, Parser)]
+pub(crate) struct Pending {}
+
+impl Pending {
+  pub(crate) fn run(self, wallet: Wallet) -> SubcommandResult {
+    let etchings = wallet
+      .pending_etchings()?
+      .into_iter()
+      .map(|(_, entry)| entry)
```
Comment (ldiego08) [inline src/subcommand/wallet/pending.rs:15]: I'm just returning the entire etching entry, making the output quite verbose. Please, let me know if another output schema is more desirable.

File: src/subcommand/wallet/pending.rs
Comment (ldiego08) [inline src/subcommand/wallet/pending.rs:1]: Not sure about the name as it doesn't specify it means pending _etchings._ Feel free to suggest a different name.

Comment (raphjaph) [reply src/subcommand/wallet/pending.rs:1]: `resume` is also kind of ambiguous so I think this should be fine. We can always change it if we find a better name.

File: src/subcommand/wallet/pending.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+use {super::*, crate::wallet::entry::EtchingEntry};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct PendingOutput {
+  pub etchings: Vec<EtchingEntry>,
+}
```
Comment (raphjaph) [inline src/subcommand/wallet/pending.rs:6]:
```suggestion
pub struct PendingOutput {
  pub rune: SpacedRune,
  pub commit: Txid,
}
```

Let's just return the Rune name and the commit Txid for now.

File: src/subcommand/wallet/pending.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+impl Pending {
+  pub(crate) fn run(self, wallet: Wallet) -> SubcommandResult {
+    let etchings = wallet
+      .pending_etchings()?
+      .into_iter()
+      .map(|(_, entry)| entry)
+      .collect();
+
+    Ok(Some(Box::new(PendingOutput { etchings }) as Box<dyn Output>))
```
Comment (raphjaph) [inline src/subcommand/wallet/pending.rs:18]:
```suggestion
    Ok(Some(Box::new(PendingOutput { etchings }) as Box<dyn Output>))
```

Down here then return a Vec<PendingOutput>

**Review by raphjaph — CHANGES_REQUESTED** [review]
Looks good so far. Definitely also needs an integration test. You could put that in the `tests/wallet/resume.rs` file.

File: src/subcommand/wallet/pending.rs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#[derive(Debug, Parser)]
+pub(crate) struct Pending {}
+
+impl Pending {
+  pub(crate) fn run(self, wallet: Wallet) -> SubcommandResult {
+    let etchings = wallet
+      .pending_etchings()?
+      .into_iter()
+      .map(|(rune, entry)| {
+        let spaced_rune = rune.to_string().parse::<SpacedRune>().unwrap();
```
Comment (ldiego08) [inline src/subcommand/wallet/pending.rs:17]: @raphjaph what's the best way to get a `SpacedRune` from a `Rune`?

Comment (raphjaph) [reply src/subcommand/wallet/pending.rs:17]: It's buried inside the entry struct. Something like this:
```
let spaced_rune = entry.output.rune.unwrap().rune
```

**Review by raphjaph — APPROVED** [review]
Nice, LGTM!

**Pull request merged**